packages/playwright-core/src/utils/isomorphic/locatorGenerators.ts

```diff
@@ -158,19 +158,29 @@ function innerAsLocators(factory: LocatorFactory, parsed: ParsedSelector, isFram
       }
     }
     if (part.name === 'internal:control' && (part.body as string) === 'enter-frame') {
-      tokens.push([factory.generateLocator(base, 'frame', '')]);
+      const lastTokens = tokens[tokens.length - 1];
+      const lastPart = parts[index - 1];
+
+      const extendedTokens = lastTokens.map(token =>
```

Comment:
This mangling needs a comment with an example of what was in `lastTokens` and what will be there.